### ISSUE TITLE
Built in warnings to help people migrate to v7?

### ISSUE THREAD

**Issue by jodeleeuw** [jspsych/jsPsych#2217]
I wonder if we could add some built in warnings to v7 that would help people migrate.

* If a `type` parameter is a string
* If `jsPsych.init()` is called
* Is there a way to warn people who try to use `window.jsPsych` without messing things up for `initJsPsych()`?

**Comment by chrisbrickhouse** [comment]
>some errors in user code may be easier to find when `window.jsPsych` is `undefined`
@bjoluc what errors do you have in mind? I might be misunderstanding something since I'm new to TypeScript, but if we assume `var jsPsych = ...` masks the module definition of `window.jsPsych`, aren't we also assuming that `window.jsPsych` is already defined?

**Comment by chrisbrickhouse** [comment]
That's very helpful! Thanks for that example. I agree with you that `jsPsych is undefined` is more clear, but I think my question is how will an end user even reach the state of calling `jsPsych.something()` without having already run `jsPsych.init()`?

To answer my own question, an [old experiment of mine](https://github.com/chrisbrickhouse/Experiment-templates/blob/master/text-norming/semantic-associations-text.js) from 6.x is probably a good example. In that experiment I construct the timeline which includes calls to `jsPsych.timelineVariable` and `jsPsych.NO_KEYS` before the `jsPsych.init()` call. 

With the current set up, the console would log a series of `jsPsych is undefined` which would be pretty clear. If we defined `window.jsPsych.init()` then the console would log `jsPsych.timelineVariable is not a function` etc and then whatever error text we defined for `window.jsPsych.init()`. 

For me I think the critical question is how useful can we make the error text and does that justify making the other console warnings less clear? The issue suggests warnings, but in this case it might be worthwhile to throw an uncaught exception. Wrongly calling `jsPsych.init()` is not only incorrect, but something that needs to be fixed before allowing any further execution of code. going beyond a warning and actually halting execution might make debugging even easier despite the less clear error logs for the other functions.
